### PR TITLE
Flow end flags/v2

### PR DESCRIPTION
**Pull request by victorjulien**
#11790 with a flag typo fixup.

### CODE REVIEW

**Comment by catenacyber** [comment]
Is there a ticket ?

**Comment by victorjulien** [comment]
> Is there a ticket ?

No, it's just a cleanup.

**Review by catenacyber — APPROVED** [review]
Looks good to me as code simplification

File: src/output-json-flow.c [hunk 1]
```diff
@@ -230,32 +230,33 @@ static void EveFlowLogJSON(OutputJsonThreadCtx *aft, JsonBuilder *jb, Flow *f)
...
-                break;
+        case FLOW_STATE_CAPTURE_BYPASSED:
+            JB_SET_STRING(jb, "state", "bypassed");
+            JB_SET_STRING(jb, "bypass", "capture");
+            break;
 #endif
-            default:
-                SCLogError("Invalid flow state: %d, contact developers", flow_state);
-        }
+        case FLOW_STATE_SIZE:
```
Comment (catenacyber) [inline src/output-json-flow.c:255]: > Remove use of `default` to help the compiler flag new enum additions.

I do not understand this change.
This is C, not rust, the compiler is ok with incomplete switch...

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .uptime | 649 | 628 | 96.76% | 

Pipeline 22721

File: src/output-json-flow.c (hunk 1 above)
Comment (victorjulien) [reply src/output-json-flow.c:255]: my default set of flags will warn

File: src/output-json-flow.c [hunk 2]
```diff
@@ -230,32 +230,33 @@ static void EveFlowLogJSON(OutputJsonThreadCtx *aft, JsonBuilder *jb, Flow *f)
...
+            DEBUG_VALIDATE_BUG_ON(1);
+            SCLogDebug("invalid flow state: %d, contact developers", flow_state);
```
[the remark this answers is not in the document]
Comment (jlucovsky) [reply src/output-json-flow.c:257]: This looks serious ... should it be ignored when debug-validation/debug are not enabled?

**Comment by victorjulien** [comment]
Merged in #11804, thanks!

**Pull request closed without merging**